### ISSUE TITLE
napalm/capirca issue

### ISSUE THREAD

**Issue by AmbicaY** [saltstack/salt#46918]
### Description of Issue/Question
netacl.load_term_config command errors out saying "'_Term' object has no attribute 'priority'
"
### Setup
salt 2017.7.3 (Nitrogen)

Installed capirca using:
`pip install -e git+git@github.com:google/capirca.git#egg=aclgen`

pillar file for the device:

```
netacl:
  my-filter:
    my-term:
      action:
        accept
      destination_address:
        any
      source_address:
        1.2.3.4/24
```

### Steps to Reproduce Issue
```
salt testdevice-ios netacl.load_term_config my-filter term_name=my-term pillar_key=netacl test=True debug=True
testdevice-ios:
    The minion function caused an exception: Traceback (most recent call last):
      File "/usr/lib/python2.7/dist-packages/salt/minion.py", line 1600, in _thread_return
        return_data = minion_instance.executors[fname](opts, data, func, args, kwargs)
      File "/usr/lib/python2.7/dist-packages/salt/executors/direct_call.py", line 12, in execute
        return func(*args, **kwargs)
      File "/usr/lib/python2.7/dist-packages/salt/utils/napalm.py", line 430, in func_wrapper
        return func(*args, **kwargs)
      File "/usr/lib/python2.7/dist-packages/salt/modules/napalm_acl.py", line 446, in load_term_config
        **term_fields)
      File "/usr/lib/python2.7/dist-packages/salt/modules/capirca_acl.py", line 869, in get_term_config
        revision_date_format=revision_date_format)
      File "/usr/lib/python2.7/dist-packages/salt/modules/capirca_acl.py", line 1015, in get_filter_config
        revision_date_format=revision_date_format)
      File "/usr/lib/python2.7/dist-packages/salt/modules/capirca_acl.py", line 1180, in get_policy_config
        merge_pillar=merge_pillar)
      File "/usr/lib/python2.7/dist-packages/salt/modules/capirca_acl.py", line 538, in _get_policy_object
        **term_fields)
      File "/usr/lib/python2.7/dist-packages/salt/modules/capirca_acl.py", line 493, in _get_term_object
        log.debug(six.text_type(term))
      File "/root/src/aclgen/lib/policy.py", line 747, in __str__
        if self.ttl:
    AttributeError: '_Term' object has no attribute 'ttl'
```

### Versions Report

```
Salt Version:
           Salt: 2017.7.3

Dependency Versions:
           cffi: 1.11.4
       cherrypy: 3.5.0
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: 2.18
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Dec  4 2017, 14:50:18)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-87-generic
         system: Linux
        version: Ubuntu 16.04 xenial
```

```
pip freeze:
absl-py==0.1.13
apache-libcloud==1.5.0
asn1crypto==0.24.0
bcrypt==3.1.4
-e git+git@github.com:google/capirca.git@e777ef53c31617dc192e4924e2f72e259a757dab#egg=capirca
cffi==1.11.4
chardet==2.3.0
CherryPy==3.5.0
cryptography==2.1.4
enum34==1.1.6
funcsigs==1.0.2
future==0.16.0
futures==3.0.5
gitdb==0.6.4
GitPython==1.0.1
idna==2.6
infoblox==1.1.1
ipaddr==2.2.0
ipaddress==1.0.19
Jinja2==2.8
jtextfsm==0.3.1
junos-eznc==2.1.7
lockfile==0.12.2
lxml==4.1.1
Mako==1.0.3
MarkupSafe==0.23
mock==2.0.0
msgpack-python==0.4.6
napalm==2.3.0
napalm-base==1.0.0
napalm-eos==0.6.1
napalm-ios==0.8.1
napalm-iosxr==0.5.6
napalm-junos==0.12.1
napalm-nxos==0.7.1
ncclient==0.5.3
netaddr==0.7.18
netmiko==2.0.1
paramiko==2.4.0
pbr==4.0.1
ply==3.7
psutil==3.4.2
pyasn1==0.1.9
pycparser==2.18
pycrypto==2.6.1
pycurl==7.43.0
pyeapi==0.8.1
pyIOSXR==0.52
PyMySQL==0.7.2
PyNaCl==1.2.1
pynxos==0.0.3
pyOpenSSL==17.5.0
pyserial==3.4
python-apt==1.1.0b1
python-dateutil==2.4.2
python-gflags==3.1.2
python-systemd==231
PyYAML==3.11
pyzmq==15.2.0
repoze.lru==0.6
requests==2.9.1
Routes==2.2
salt==2017.7.3
scp==0.10.2
simplejson==3.8.1
six==1.10.0
smmap==0.9.0
textfsm==0.3.2
tornado==4.2.1
urllib3==1.13.1
WebOb==1.5.1
```

Salt minion proxy:

```
salt-proxy 2017.7.3 (Nitrogen)
root@sc-net-salt-proxy-dev-01:~# pip freeze
absl-py==0.1.13
asn1crypto==0.24.0
bcrypt==3.1.4
-e git+git@github.com:google/capirca.git@e777ef53c31617dc192e4924e2f72e259a757dab#egg=capirca
cffi==1.11.4
chardet==2.3.0
cryptography==2.1.4
enum34==1.1.6
funcsigs==1.0.2
future==0.16.0
futures==3.0.5
idna==2.6
ipaddr==2.2.0
ipaddress==1.0.19
Jinja2==2.8
jtextfsm==0.3.1
junos-eznc==2.1.7
lxml==4.1.1
Mako==1.0.3
MarkupSafe==0.23
mock==2.0.0
msgpack-python==0.4.6
napalm==2.3.0
napalm-base==1.0.0
napalm-eos==0.6.1
napalm-ios==0.8.1
napalm-iosxr==0.5.6
napalm-junos==0.12.1
napalm-nxos==0.7.1
ncclient==0.5.3
netaddr==0.7.19
netmiko==2.0.1
paramiko==2.4.0
pbr==4.0.1
ply==3.7
psutil==3.4.2
pyasn1==0.1.9
pycparser==2.18
pycrypto==2.6.1
pycurl==7.43.0
pyeapi==0.8.1
pyIOSXR==0.52
PyMySQL==0.7.2
PyNaCl==1.2.1
pynxos==0.0.3
pyOpenSSL==17.5.0
pyserial==3.4
python-apt==1.1.0b1
python-dateutil==2.4.2
python-gflags==3.1.2
python-systemd==231
PyYAML==3.11
pyzmq==15.2.0
requests==2.9.1
salt==2017.7.3
scp==0.10.2
six==1.10.0
textfsm==0.3.2
tornado==4.2.1
urllib3==1.13.1
```

**Comment by Ch3LL** [comment]
ping @mirceaulinic any chance you can take a look here?

**Comment by mirceaulinic** [comment]
Hey @AmbicaY would you be able to upgrade to 2018.3.0? This should be already fix in Oxygen.

**Comment by AmbicaY** [comment]
@mirceaulinic, Could you please check this issue?

**Comment by AmbicaY** [comment]
@mirceaulinic @Ch3LL 

Can you guys take a look at this issue? I could repro this issue on another salt installation also.

**Comment by mirceaulinic** [comment]
Hey @AmbicaY - I think I see what's the issue: looks like `ttl` is a new field defined in the underlying library Capirca, and we should should apply a small patch to the `capirca_acl` module, similar to the changes from https://github.com/saltstack/salt/pull/44508/files.
I won't be able to get to it before next week, but if you can please go ahead and port the module into your `file_roots` (i.e., copy https://github.com/saltstack/salt/blob/develop/salt/modules/capirca_acl.py into the `_modules` directory, apply the change), then a PR would be greatly appreciated, so the fix will be available into 2018.3.1. Thanks!

**Comment by AmbicaY** [comment]
Hi @mirceaulinic 

I think you have added the priority field and not the ttl field. Please check.

**Comment by mirceaulinic** [comment]
That's correct - the other PR added the priority field as that one was
missing back then, while to solve this issue we'd need to add the ttl
field, in a similar manner.

**Comment by mirceaulinic** [comment]
This should be fixed via https://github.com/saltstack/salt/commit/6135b76e2cd28f9d5a550896808c2de6970267af.

@AmbicaY Till 2018.3.1 is out, you can copy the code in your own environment.

**Comment by rallytime** [comment]
Closed via #47202
